cloud_AWS/terraform/module/examples/multiple-vpc-regions/aws_multiregions.py

```diff
@@ -0,0 +1,174 @@
+import argparse
+import logging
+import json
+import os
+import sys
+from typing import Any, Callable, Dict, Tuple
+
+from python_terraform import IsFlagged, IsNotFlagged, Terraform
+
+log = logging.getLogger(__name__)
+logging.basicConfig(
+    format="%(asctime)s [%(name)s] %(levelname)s: %(message)s", level=logging.INFO, filename="multiregions.log"
+)
+
+EX_OK: int = 0  # exit code for successful command
+EX_FAILED: int = 1  # exit  code for failed command
+
+DFAULT_VPC_REGIONS_FILE_NAME: str = "input_data.json"
+
+TerraformVars = Dict[str, Any]  # variables passed in terraform plan/apply/destroy call
+TerraformAction = Callable[[Terraform, TerraformVars], bool]  # terraform plan/apply/destroy
+
+
+def execute_action(action: TerraformAction, vpc_regions: dict) -> bool:
+    """Execute an action for every region"""
+
+    t_bucket = Terraform(working_dir="bucket/")
+    bucket_region_name = []
+    bucket_arn_list = []
+    err_occur = False
+
+    # invoke action for each region on terraform configuration located in bucket/
+    for region in vpc_regions["regions"]:
+        print_log(f'Region: {region}, vpc ids: {vpc_regions["regions"][region]["vpc_id_list"]}')
+        tf_vars = {
+            "region": region,
+            "vpc_id_list": vpc_regions["regions"][region]["vpc_id_list"],
+            "s3_bucket_prefix": vpc_regions["regions"][region].get("s3_bucket_prefix", "")
+        }
+
+        if not (prepare_workspace(t_bucket, region) and action(t_bucket, tf_vars)):
+            err_occur = True
+
+        if t_bucket.output().get("kentik_bucket_name") and t_bucket.output()["kentik_bucket_name"].get("value"):
+            for bucket_name in t_bucket.output()["kentik_bucket_name"]["value"]:
+                bucket_region_name.append([region, bucket_name])
+        if t_bucket.output().get("kentik_bucket_arn_list") and t_bucket.output()["kentik_bucket_arn_list"].get("value"):
+            bucket_arn_list.extend(t_bucket.output()["kentik_bucket_arn_list"]["value"])
+
+    # invoke action on terraform configuration located in cloud_iam/
+    t_cloud_aim = Terraform(working_dir="cloud_iam/")
+    tf_vars = {
+        "region": list(vpc_regions["regions"].keys())[0],  # used only in provider region
+        "plan_id": vpc_regions.get("plan_id", ""),
```

Comment:
without plan_id, the API will return error, I think we can check the value and handle the case in advance

Reply:
Implemented suggestion